src/main/java/core/basesyntax/CheckElectricCar.java

```diff
@@ -1,8 +1,10 @@
 package core.basesyntax;
 
-/**
- * <p>Имплементируйте необходимый функциональный интерфейс и его метод, который реализует проверку
- * установлен ли на авто EngineType.ELECTRIC. Метод должен принимать Car.</p>
- */
-public class CheckElectricCar {
+public class CheckElectricCar implements CheckElectric {
+
+    @Override
+    public boolean test(Car car) {
+        EngineType engineType = car.getEngine().getEngineType();
+        return engineType == EngineType.ELECTRIC;
```

Comment:
Я думаю лучше объекты через equals сравнивать)

Reply:
да, спасибо, может так корректнее раз equals переопределен. done

Reply:
@aleksandr-hrankin unlike comparing String in Java, you can use both == and equals() method to compare Enum, they will produce same result because equals() method of Java.lang.Enum internally uses == to compare enum in Java.

Reply:
Понял, принял, спасибо!